### PR TITLE
Removed gotcha from user registration process #3168

### PR DESCRIPTION
**Pull request by trystant**
In your PR did you:
Removed gotcha from the user registration process because it doesn't work with Rails 5 and Devise. Provides a short term fix for 3168

  - [X] Include a description of the changes?
  - [X] Mention the issue the PR addresses?
  - [ ] Include screenshots of any changes to the UI?
  - [X] Isolate any changes to gems (meaning that any new, updated, or removed gems and resulting code changes should be in their own PR)?
  - [ ] Add and/or update specs for your code?

### CODE REVIEW

File: spec/requests/registrations_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+RSpec.describe 'Registrations', type: :request do
```
Comment (rlgreen91) [inline spec/requests/registrations_spec.rb]: I would prefer this match the formatting of the other request specs.

File: spec/requests/registrations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  describe 'POST /registrations' do
+    it 'works! (now write some real specs)' do
+      get '/'
+      expect(response).to render_template('index')
```
Comment (rlgreen91) [inline spec/requests/registrations_spec.rb]: I don't see a need for this assertion.

Comment (trystant) [reply spec/requests/registrations_spec.rb]: It was to confirm a request beforehand...but I can keep things to single spec per request.

File: spec/requests/registrations_spec.rb (hunk 1 above)
Comment (trystant) [reply spec/requests/registrations_spec.rb]: Can you be specific? I modeled this after `spec/requests/organizations_spec.rb`

Comment (rlgreen91) [reply spec/requests/registrations_spec.rb]: Ah, it wasn't clear to me what the test was focused on before, but this looks good.

**Comment by trystant** [comment]
> I still want to know why we're adding this gem back.

That gem is required for the `render_template` method used in the spec.

**Comment by rlgreen91** [comment]
> > I still want to know why we're adding this gem back.
> 
> That gem is required for the `render_template` method used in the spec.

I see.  I'm not a fan of checking that the template is rendered in my request specs - I'd just test that some string in the body is present like you're doing now and leave it at that.  In my opinion, a request spec shouldn't really be concerned about what the response looks like in terms of HTML, just that the request was processed properly, and in this case that's including an error message in the body.  I don't particularly need to know that it rendered a specific template - that's what the view specs are for.

**Comment by trystant** [comment]
I've read through documentation discussing the recommendation to remove `rails-controller-testing` and I don't think I agree with it, but my objection isn't strong enough to stop things at this point in time.

Why? I feel like it's helpful to know whether the request with params goes to the correct page. `render_template` was the mechanism that request specs used to determine that the correct page was rendered at the end of the request. Seems like it's possible to render a page that has the correct status, but incorrect location.

Using `pry` I noticed that both `response.uri` and `response.location` are `nil` in these tests, so that would probably have to be changed in order to make the expectation I'm looking for possible.

We'll see how it works though. I've updated specs and removed that gem from the build for now.

**Pull request merged**